completions/doas

```diff
@@ -0,0 +1,45 @@
+# doas(1) completion                                       -*- shell-script -*-
+
+_comp_cmd_doas()
+{
+    local cur prev words cword split
+    _init_completion -s || return
+
+    local i
+
+    for ((i = 1; i <= cword; i++)); do
+        if [[ ${words[i]} != -* ]]; then
+            local PATH=$PATH:/sbin:/usr/sbin:/usr/local/sbin
+            local root_command=${words[i]}
+            _command_offset $i
```

Comment:
In a similar way to`_comp_root_command()` (defined in `bash_completion`), I think we want to enable the completion as if the current user has root privilege.

```suggestion
            local _comp_root_command=1
            _command_offset $i
```